testutil/testutil.go

```diff
@@ -73,16 +80,64 @@ func GetAvailableLocalAddress(t *testing.T) string {
 // available for opening when this function returns provided that there is no
 // race by some other code to grab the same port immediately.
 func GetAvailablePort(t *testing.T) uint16 {
-	endpoint := GetAvailableLocalAddress(t)
-	_, port, err := net.SplitHostPort(endpoint)
-	require.NoError(t, err)
+	var exclusions []portpair
+	portFound := false
+	var port string
+	var err error
+	if runtime.GOOS == "windows" {
+		exclusions = getExclusionsList(t)
+	}
+
+	for !portFound {
+		endpoint := GetAvailableLocalAddress(t)
+		_, port, err = net.SplitHostPort(endpoint)
+		require.NoError(t, err)
+		portFound = true
+		if runtime.GOOS == "windows" {
+			for _, pair := range exclusions {
+				if port >= pair.first && port <= pair.last {
+					portFound = false
+					break
+				}
+			}
+		}
+	}
```

Comment:
Please add a comment to explain why this is needed. You have a good description in the PR, so perhaps reuse the bits from it.